### PR TITLE
44077 Fix to QA bug found by Francis where the publish template will …

### PR DESCRIPTION
**Pull request by gomezll**
…fail to apply if the geo name has non-alphanumeric characters in it. Stripping non-alphanum characters out of the geo name just before publish

### CODE REVIEW

File: hooks/tk-multi-publish2/basic/publish_mari_textures.py [hunk 1]
```diff
@@ -248,12 +249,20 @@ def publish(self, settings, item):
         ctx_fields = self.parent.context.as_template_fields(publish_template)
         fields.update(ctx_fields)
 
-        version = 0
+        # For geo name, strip out the non-alphanumeric characters because Mari's
+        # publish template filter does not allow non-alphanumeric characters in
+        # the geo name.
+        pattern = re.compile('[\W_]+', re.UNICODE)
```
Comment (josh-t) [inline hooks/tk-multi-publish2/basic/publish_mari_textures.py]: I'd change these two lines to be:

```python
geo_name = re.sub(r"[\W_]+", "", geo_name)
```

* Do we need `re.UNICODE`?
* Double quotes to align with Tk standards

File: hooks/tk-multi-publish2/basic/publish_mari_textures.py [hunk 2]
```diff
@@ -248,12 +249,20 @@ def publish(self, settings, item):
...
+        geo_name = pattern.sub('', geo_name)
         fields["name"] = geo_name
-        fields["channel"] = channel_name
+
+        version = 0
```
Comment (josh-t) [inline hooks/tk-multi-publish2/basic/publish_mari_textures.py]: Why are we hardcoding version 0?

File: hooks/tk-multi-publish2/basic/publish_mari_textures.py (hunk 1 above)
Comment (gomezll) [reply hooks/tk-multi-publish2/basic/publish_mari_textures.py]: Made changes you requested

File: hooks/tk-multi-publish2/basic/publish_mari_textures.py (hunk 2 above)
Comment (gomezll) [reply hooks/tk-multi-publish2/basic/publish_mari_textures.py]: Reorganized code so that this isn't necessary.

**Pull request merged**